src/keywordPreservation.ts

```diff
@@ -91,7 +91,7 @@ export const getAssetsValueByAddress = (address: string, assets: any): any => {
 
   // It is easier to handle an address piece-by-piece by
   // splitting on the period into separate "directions"
-  const directions = address.split('.');
+  const directions = address.split(/\.(?![^\[]*\])/g);
```

Comment:
Splits periods unless they're within square brackets `[]`. This is because with this new functionality, the address may have a URL or value with a period in it. Example: `customDomains.[domain=travel0.com].domain`